### PR TITLE
Update API documents for splitter and transformer

### PR DESCRIPTION
**Pull request by nd-02110114**
I updated API documents for splitter and transformer.

## What I did

First, I fixed the docstrings. The main modofication is converting `Notes` to `Note`.
`Note` renders better style like this.

スクリーンショット 2020-11-05 23 38 41

The current docs

スクリーンショット 2020-11-05 23 38 59

And then, I added local table of contents like this.

スクリーンショット 2020-11-05 23 37 59

### CODE REVIEW

File: deepchem/trans/transformers.py
Comment (peastman) [inline deepchem/trans/transformers.py:1580]: Are those really the only ones it works with?  I looked at the code and didn't see any reason it wouldn't work with an ImageDataset (or other dataset classes we might create in the future).

File: docs/source/api_reference/splitters.rst
Comment (rbharath) [inline docs/source/api_reference/splitters.rst:112]: s/FingeprintSplitter/FingerprintSplitter

File: docs/source/api_reference/transformers.rst
Comment (rbharath) [inline docs/source/api_reference/transformers.rst:11]: We should keep the references for the abstract `Transformer` superclass around. It's a useful guide for developers

File: docs/source/api_reference/transformers.rst
Comment (rbharath) [inline docs/source/api_reference/transformers.rst:98]: Should these be removed?

**Review by rbharath — COMMENTED** [review]
Thanks for kicking this off! Have a few comments to start us off

File: docs/source/api_reference/splitters.rst
Comment (nd-02110114) [reply docs/source/api_reference/splitters.rst]: I fixed.

File: docs/source/api_reference/transformers.rst
Comment (nd-02110114) [reply docs/source/api_reference/transformers.rst:11]: I fixed.

File: deepchem/trans/transformers.py
Comment (nd-02110114) [reply deepchem/trans/transformers.py:1627]: This is a return type. I seem that this function returns DiskDataset or NumpyDataset.

File: docs/source/api_reference/transformers.rst
Comment (nd-02110114) [reply docs/source/api_reference/transformers.rst:98]: `FeaturizationTransformer` just moved.

スクリーンショット 2020-11-17 9 40 15

**Comment by nd-02110114** [comment]
Sorry for the late fix. I think this PR is ready to a second review.

File: deepchem/splits/splitters.py
Comment (rbharath) [inline deepchem/splits/splitters.py:405]: Ah nice fix!

File: deepchem/trans/transformers.py
Comment (rbharath) [inline deepchem/trans/transformers.py:149]: Nice renaming here!

**Review by rbharath — APPROVED** [review]
Looks good for me!

Holding off on merging in case @peastman has further comments, but otherwise good to go

**Comment by peastman** [comment]
Looks fine to me!

**Comment by nd-02110114** [comment]
Thanks you for reviews! I merge in.

**Pull request merged**
